proposals/solid-oidc/primer.bs

```diff
@@ -0,0 +1,745 @@
+
+```
+9XmwK8mQ3H5-PnzAt3lFHzWBW_v5QhYynezbbit4kC8
+```
+
+#### 21. Generates access token
+
+The OS generates an access token. To do so it constructs a JSON Web Token and signs it using its own keys. The access token could look like (remember you can decode it using https://jwt.io):
+
+```
+eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJodHRwczovL2FsaWNlLmNvb2xwb2QuZXhhbXBsZS9wcm9maWxlL2NhcmQjbWUiLCJ3ZWJpZCI6Imh0dHBzOi8vYWxpY2UuY29vbHBvZC5leGFtcGxlL3Byb2ZpbGUvY2FyZCNtZSIsImlzcyI6Imh0dHBzOi8vc2VjdXJlYXV0aC5leGFtcGxlIiwiYXVkIjoic29saWQiLCJjbmYiOnsiamt0IjoiOVhtd0s4bVEzSDUtUG56QXQzbEZIeldCV192NVFoWXluZXpiYml0NGtDOCJ9LCJjbGllbnRfaWQiOiJodHRwczovL2RlY2VudHBob3Rvcy5leGFtcGxlL3dlYmlkI3RoaXMiLCJqdGkiOiJmZDBhMTM1My0yNWYzLTRjN2UtOGY4Yi1jMTQyNjRmMWMxMmUiLCJpYXQiOjE2MDMzODUyNjEsImV4cCI6MTYwMzM4NTg2MX0.HIIYiovILPvsdkD3s3xomR1MkA_Ir8Mx_C-eHxbWEnw9Z2rv2rOWbvMPZ--BqH3qVSupgYTZZsTDbZXA8giCLA
+```
+
+Token Header:
+```
+{
+  "alg": "ES256",
+  "typ": "JWT"
+}
+```
+ - `"alg": "ES256"`: indicates the token was signed using eliptic curve
+ - `"typ": "JWT"`: indicates that this is a JSON web token
+
+Token Body:
+```
+{
+  "sub": "https://alice.coolpod.example/profile/card#me",
+  "webid": "https://alice.coolpod.example/profile/card#me",
+  "iss": "https://secureauth.example",
+  "aud": "solid",
+  "cnf": {
+    "jkt": "9XmwK8mQ3H5-PnzAt3lFHzWBW_v5QhYynezbbit4kC8"
+  },
+  "client_id": "https://decentphotos.example/webid#this",
+  "jti": "fd0a1353-25f3-4c7e-8f8b-c14264f1c12e",
+  "iat": 1603385261,
+  "exp": 1603385861
+}
+```
+ - `"sub": "https://alice.coolpod.example/profile/card#me"`: The user this token corresponds to. (their WebId)
+ - `"webid": "https://alice.coolpod.example/profile/card#me"`: The WebId of the user that logged in
+ - `"iss": "https://secureauth.example"`: The OP that was used to generate this token
+ - `"aud": "solid"`: The token's audience. Because DPoP handles scoping the audience in all practical senses, the audience for a solid token is simply "solid."
+ - `"cnf": { "jkt": "9XmwK8mQ3H5-PnzAt3lFHzWBW_v5QhYynezbbit4kC8" }`: The jwk thrumbprint must be embedded in an object on the field "jkt"
+ - `"client_id": "https://decentphotos.example/webid#this"`: The client id of the application
+ - `"iat": 1603370641`: The date the token was issued, in this case October 22, 2020 8:44:01
+ - `"exp": 1603371241`: The token's expiration date, in this case October 22, 2020 8:54:01
+
+#### 22. Generates id token
+
+If `id_token` was included in the authorization request, generate the id_token again just as you generated it in step 10.
+
+#### 23. Generates refresh token
+
+If `offline_access` was provided as a scope, the OP creates a cryptographically random long as a refresh token. It could be like the one below. Notice the one below is a JWT, but a refresh token does not need to be a JWT.
```

Comment:
"cryptographically random long"? This is merely an opaque token from the perspective of the client, which can be exchanged for access tokens in the future.